geyser-plugin-manager/src/geyser_plugin_service.rs

```diff
@@ -208,4 +221,384 @@ impl GeyserPluginService {
         self.plugin_manager.write().unwrap().unload();
         Ok(())
     }
+
+    pub fn plugin_manager_join_handle(&self) -> Arc<JoinHandle<()>> {
+        // This is only called after the initialization of a solana-test-validator or a solana-validator
+        // which has a geyser plugin service
+        Arc::clone(self.rpc_handler_thread.as_ref().unwrap())
+    }
+
+    pub fn list_plugins_rpc(plugin_manager: &GeyserPluginManager) -> JsonRpcResult<Vec<String>> {
+        Ok(plugin_manager
+            .plugins
+            .iter()
+            .map(|p| p.name().to_owned())
+            .collect())
+    }
+
+    pub fn load_plugin_rpc(
+        plugin_manager: &mut GeyserPluginManager,
+        config_file: &str,
+    ) -> JsonRpcResult<String> {
+        // Load plugin
+        let mut new_plugin: Box<dyn GeyserPlugin> = load_plugin_from_config(Path::new(config_file))
+            .map_err(|err| jsonrpc_core::Error {
+                code: ErrorCode::InternalError,
+                message: err.to_string(),
+                data: None,
+            })?;
+
+        // Then see if a plugin with this name already exists. If so, abort
+        if plugin_manager
+            .plugins
+            .iter()
+            .any(|plugin| plugin.name().eq(new_plugin.name()))
+        {
+            return Err(jsonrpc_core::Error {
+                code: ErrorCode::InternalError,
+                message: format!(
+                    "There already exists a plugin {} loaded. Did not load requested plugin",
+                    new_plugin.name()
+                ),
+                data: None,
+            });
+        }
+
+        // Call on_load and push plugin
+        new_plugin
+            .on_load(config_file)
+            .map_err(|on_load_err| jsonrpc_core::Error {
+                code: ErrorCode::InternalError,
+                message: format!("on_load method of plugin failed: {on_load_err}",),
+                data: None,
+            })?;
+        let name = new_plugin.name().to_string();
+        plugin_manager.plugins.push(new_plugin);
+
+        Ok(name)
+    }
+
+    pub fn unload_plugin_rpc(
+        plugin_manager: &mut GeyserPluginManager,
+        name: &str,
+    ) -> JsonRpcResult<()> {
+        // Check if any plugin names match this one
+        let Some(idx) = plugin_manager.plugins.iter().position(|plugin| plugin.name().eq(name)) else {
```

Comment:
As mentioned do not directly modify the plugins under plugin manager -- these logic belongs to plugin manager.